### PR TITLE
addition of buf.gen and proto readme file

### PR DESCRIPTION
**Pull request by joshLong145**
* addition of buf.gen file for ts and js targets

## Checklist

* [ ] Unit tests
* [ ] Documentation

## References *(optional)*

Fixes
Connects

 Mention [stepsize] in a comment if you'd like to report some technical debt. See examples here.

### CODE REVIEW

**Review by prdlk — CHANGES_REQUESTED** [review]
Lets add to CI/CD for publishing to buf.build so this process can be automated by pulling from the hosted registry

**Comment by joshLong145** [comment]
> Lets add to CI/CD for publishing to buf.build so this process can be automated by pulling from the hosted registry

I was going to introduce this in a seperate pr. but ill just add it to this one

**Review by mcjcloud — APPROVED** [review]

**Review by ntindle — APPROVED** [review]

File: proto/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+## Installation
+- Install [buf.build cli](https://buf.build/)
+- Install [protoc](https://grpc.io/docs/protoc-installation/)
+
+these installations will be enough local generation of `Go` definitions and publishing to the Sonr `buf registry`
+## Building JS/TS definitions
+For target TypeScript and JavaScript we will need `protoc` installed and the [](https://www.npmjs.com/package/protoc-gen-ts)
+
+### building proto defs
+- in the root of `proto` directory run `buf generate`
```
Comment (ntindle) [inline proto/README.md:14]: Make sure this exists in README.md as a prerequisite and that the install guide still works

**Review by ntindle — CHANGES_REQUESTED** [review]

File: proto/README.md (hunk 1 above)
Comment (prdlk) [reply proto/README.md:14]: @ntindle this is for buf.build docs right here https://buf.build/sonr-io/blockchain - lets also add redirect to our docs on this along with our license @joshLong145

**Pull request closed without merging**